src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -54,20 +60,157 @@ class Point(int x, int y);
             );
 
-            comp = CreateCompilation(src1, parseOptions: TestOptions.RegularPreview);
+            comp = CreateCompilation(src1);
             comp.VerifyDiagnostics(
                 // (2,12): error CS8761: Records must have both a 'data' modifier and parameter list
                 // class Point(int x, int y);
                 Diagnostic(ErrorCode.ERR_BadRecordDeclaration, "(int x, int y)").WithLocation(2, 12)
             );
-            comp = CreateCompilation(src2, parseOptions: TestOptions.RegularPreview);
+            comp = CreateCompilation(src2);
             comp.VerifyDiagnostics(
                 // (2,12): error CS8761: Records must have both a 'data' modifier and parameter list
                 // data class Point { }
                 Diagnostic(ErrorCode.ERR_BadRecordDeclaration, "Point").WithLocation(2, 12)
             );
-            comp = CreateCompilation(src3, parseOptions: TestOptions.RegularPreview);
+            comp = CreateCompilation(src3);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void RecordProperties_01()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+1
+2");
+        }
+
+        [Fact]
+        public void RecordProperties_02()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public C(int a, int b)
+    {
+    }
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            var comp = CreateCompilation(src);
+            comp.VerifyDiagnostics(
+                // (3,13): error CS8762: There cannot be a primary constructor and a member constructor with the same parameter types.
+                // data class C(int X, int Y)
+                Diagnostic(ErrorCode.ERR_DuplicateRecordConstructor, "(int X, int Y)").WithLocation(3, 13)
+            );
+        }
+
+        [Fact]
+        public void RecordProperties_03()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public int X { get; }
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+0
+2");
+        }
+
+        [Fact]
+        public void RecordProperties_04()
+        {
+            var src = @"
+using System;
+data class C(int X, int Y)
+{
+    public int X { get; } = 3;
+
+    public static void Main()
+    {
+        var c = new C(1, 2);
+        Console.WriteLine(c.X);
+        Console.WriteLine(c.Y);
+    }
+}";
+            CompileAndVerify(src, expectedOutput: @"
+3
```

Comment:
We should bring this behavior up in LDM. This is certainly not what I was expecting to happen in this class declaration. I'd expect either an error, or the `X`s to line up.